mail_tracking/static/src/js/failed_message/discuss.js

```diff
@@ -0,0 +1,398 @@
+                    .addClass('d-none');
+            }
+
+            this._super.apply(this, arguments);
+        },
+
+        /**
+         * Overrides to update 'set all as reviewed' button
+         *
+         * @Override
+         */
+        _updateButtonStatus: function (disabled, type) {
+            if (this._thread.getID() === 'mailbox_failed') {
+                this.$buttons
+                    .find('.o_mail_discuss_button_set_all_reviewed')
+                    .toggleClass('disabled', disabled);
+                //  Display Rainbowman when all inbox messages are reviewed
+                // through 'TOGGLE TRACKING STATUS' or marking last failed
+                // message as reviewed
+                if (disabled && type === 'toggle_tracking_status') {
+                    this.trigger_up('show_effect', {
+                        message: _t(
+                            "Congratulations, your failed mailbox is empty"),
+                        type: 'rainbow_man',
+                    });
+                }
+            }
+        },
+
+        /**
+         * Overrides to update messages in 'failed' mailbox thread
+         *
+         * @Override
+         */
+        _onMessageUpdated: function (message, type) {
+            var self = this;
+            var currentThreadID = this._thread.getID();
+            if (currentThreadID === 'mailbox_failed' && !message.isFailed()) {
+                this._thread.fetchMessages(this.domain)
+                    .then(function () {
+                        var options = self._getThreadRenderingOptions();
+                        self._threadWidget.removeMessageAndRender(
+                            message.getID(), self._thread, options)
+                            .then(function () {
+                                self._updateButtonStatus(
+                                    !self._thread.hasMessages(), type);
+                            });
+                    });
+            } else {
+                //  Workaround to avoid call '_fetchAndRenderThread' and refetch
+                // thread messages.
+                this._super.apply(this, arguments);
+            }
+        },
+
+        /**
+        *  Overrides to hide reply in the 'failed' mailbox, as this feature does
+        * not necessary in that channel. It also show retry and
+        * 'set as reviewed'.
+        *
+        * @Override
+        */
+        _getThreadRenderingOptions: function () {
+            var values = this._super.apply(this, arguments);
+            if (this._thread.getID() === 'mailbox_failed') {
+                values.displayEmailIcons = true;
+                values.displayReplyIcons = false;
+                values.displayRetryButton = true;
+                values.displayReviewedButton = true;
+            }
+            return values;
+        },
+
+        /**
+        * Overrides to listen event that refresh thread messages
```

Comment:
As you might have guessed, you don't need to repeat you're overriding a method when overriding it. The `@override` flag is enough to know that:

```suggestion
        * Listen also to the event that refreshes thread messages
```